src/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -3041,7 +3069,7 @@ bool QgsGdalProvider::remove()
     mGdalDataset = nullptr;
 
     CPLErrorReset();
-    CPLErr err = GDALDeleteDataset( driver, dataSourceUri().toUtf8().constData() );
+    CPLErr err = GDALDeleteDataset( driver, dataSourceUri( true ).toUtf8().constData() );
     if ( err != CPLE_None )
     {
       QgsLogger::warning( "RasterIO error: " + QString::fromUtf8( CPLGetLastErrorMsg() ) );
```

Comment:
Can you check that the credentials don't get exposed in the logs in case of failure?

Reply:
@elpaso , I've made sure none of the QgsDebugMsg were calling dataSourceUri( bool ), what else should I do.

Reply:
I was worried by `CPLGetLastErrorMsg()`

Reply:
@elpaso , what's the alternative here, not appending CPLGetLastErrorMsg() to the warning message?

Reply:
Sorry: I don't know what's in `CPLGetLastErrorMsg()`, but if there is a risk of leaking the credentials I would suggest to either try to replace them with `************` or if it's not possible to replace the native error string completely with a generic one.

Reply:
@elpaso , I've checked a couple of CPLGetLastErrorMsg() using authcfg-based URIs, credentials aren't leaked in any of the messages I've seen.